server/midddleware/quota.go

```diff
@@ -0,0 +1,30 @@
+package middleware
+
+import (
+	"context"
+
+	"github.com/tigrisdata/tigris/server/quota"
+	"github.com/tigrisdata/tigris/server/request"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/proto"
+)
+
+func quotaUnaryServerInterceptor() grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		ns, _ := request.GetNamespace(ctx)
+		if err := quota.Check(ctx, ns, proto.Size(req.(proto.Message))); err != nil {
+			return nil, err
+		}
+		return handler(ctx, req)
+	}
+}
+
+func quotaStreamServerInterceptor() grpc.StreamServerInterceptor {
```

Comment:
If I am streaming 1000 documents and I hit the quota on 100th document, what will be the behavior here?

Reply:
This is per request quota. There is no read throughput quota. We can add it later.